scipy/stats/_morestats.py

```diff
@@ -3890,3 +3890,68 @@ def circstd(samples, high=2*pi, low=0, axis=None, nan_policy='propagate', *,
     if not normalize:
         res *= (high-low)/(2.*pi)  # [1] (2.3.14) w/ (2.3.7)
     return res
+
+#@_axis_nan_policy_factory(
+#    lambda x: x, result_to_tuple=lambda x: (x,), n_outputs=1, default_axis=None
+#)
+def directionalmean(samples, *, axis=0, nan_policy='propagate'):
+    """Computes the directional mean of a sample of vectors.
+
+    Serves as equivalent of the sample mean for directional data whose
+    magnitude is irrelevant, e. g. unit vectors.
+
+    Parameters
+    ----------
+    samples : array_like
+        Input array. Must be at least two-dimensional.
```

Comment:
Please mention the convention for the axis corresponding with the dimensionality of the vector space. (Last axis, I think?)

Reply:
Currently yes. Makes me think though if this is a very friendly API in the long run. 

Suppose I have a 4D array of shape `(n, f1, coordinates, f2)` where `n` is the number of samples, `f1`, `f2` some other features and `coordinates` are points on the regular sphere $S^2 (Github Latex: where are you?). How would I now tell the function to directionally average over the `n` samples in axis 2? Not sure how other `stats` functions handle this case or if I am overcomplicating the issue.

Reply:
> I added the closing `$` to your S2 and it rendered fine for me.

Missing $$ around math is a clear sign how long I have been out of academia now.. :D

> There are two kinds of axes that code needs to know about:
> 
> * The one that gets reduced (sometimes we say it's the one along which "the statistic is taken"; here's it's the one of length `n` with independent samples). This one is specified with the `axis` argument.
> * The one corresponding with the dimensionality of the space.
> 
> There are no functions that I'm aware of that accept an argument to specify the axis corresponding with the dimension of the space. In lieu of one, here are the most closely related cases in SciPy that I can think of.
> 
> * Although they are somewhat inconsistent, the consensus for multivariate distributions seems to be that the axes corresponding with dimensionality of the space should be at the end.
> * `spearmanr` is the only reduction statistic I know of that is both multivariate (see [DEP: stats.spearmanr: deprecate existing behavior with 2D inputs #16133 (comment)](https://github.com/scipy/scipy/pull/16133#issuecomment-1166372191)) and has an `axis` argument. It only accepts 2D arrays. You can specify the axis to be reduced with `axis`, and the other is considered to be the dimensionality of the space.
> 
> I'd suggest that for this function (and any multivariate statistics to come), the dimensionality of the space should be the last ones other than those specified in `axis`. (Although I do not agree with `spearmanr`'s interface overall, this is a generalization of its behavior when only `a` is passed in.)
> 
> Under this convention, you would not pass in an array `x` with shape `(n, f1, coordinates, f2)`. In that case, I would suggest that the user call `x = np.moveaxis(x, -2, -1)` before calling `directionalmean(x)`, where they can rely on the default `axis=0`. Under the same convention, if they were feeling wild, they could also pass in `x` with shape `(m, f1, f2, coordinates, n)` where `m` and `n` are to be combined before the statistic is taken along the single, raveled axis. In this case, the user would pass `directionalmean(x, axis=(0, -1))`.
> 
> Don't worry about `tuple` axes for now. One tip to help with implementation is just call `x = np.moveaxis(x, axis, 0)` at the beginning, and then you'll always know that the reduction axis is at the beginning and the dimensionality axis is left at the end.

Ok, then I will conform to this de-facto standard. Thanks again for your always helpful answers!